weave/artifact_wandb.py

```diff
@@ -908,3 +914,46 @@ def parse(cls, uri: str) -> "WeaveWBLoggedArtifactURI":
 
     def to_ref(self) -> WandbArtifactRef:
         return WandbArtifactRef.from_uri(self)
+
+
+# This is a wrapper around an artifact that acts like a list of files.
+# It fetchs a file from the manifest on __getItem__ and can return a count without fetching all files
+@dataclasses.dataclass
+class FilesystemArtifactFileIterator(list[artifact_fs.FilesystemArtifactFile]):
+    data: list[str]
+    artifact: WandbArtifact
+    idx: int = 0
+
+    @property
+    def type(self) -> types.Type:
+        return types.List(artifact_fs.FilesystemArtifactFileType())
+
+    def __init__(self, artifact: WandbArtifact, data: list[str] = []):
+        self.data = data if len(data) > 0 else artifact._get_file_paths()
+        self.artifact = artifact
+        self.idx = 0
+
+    def __getitem__(self, key):
+        if isinstance(key, slice):
```

Comment:
it might be cleaner to do 
```
path_or_paths = self.data[key]
if isisntance(path, str):
   ...
elif isisntance(path, list):
   ...
else:
   ... err
```